zerver/lib/upload.py

```diff
@@ -374,19 +373,18 @@ def check_upload_within_quota(realm: Realm, uploaded_file_size: int) -> None:
         raise RealmUploadQuotaError(_("Upload would exceed your organization's upload quota."))
 
 
-def get_file_info(request: HttpRequest, user_file: UploadedFile) -> Tuple[str, Optional[str]]:
+def get_file_info(user_file: UploadedFile) -> Tuple[str, Optional[str]]:
 
     uploaded_file_name = user_file.name
     assert uploaded_file_name is not None
-    content_type = request.GET.get("mimetype")
-    if content_type is None:
+
+    content_type = user_file.content_type
+    # It appears Django's UploadedFile.content_type defaults to an empty string,
+    # even though the value is documented as `str | None`. So we check for both.
+    if content_type is None or content_type == "":
         guessed_type = guess_type(uploaded_file_name)[0]
         if guessed_type is not None:
             content_type = guessed_type
-    else:
-        extension = guess_extension(content_type)
-        if extension is not None:
-            uploaded_file_name = uploaded_file_name + extension
 
     uploaded_file_name = urllib.parse.unquote(uploaded_file_name)
 
```

Comment:
Do we want this to return `None` or `""` as a content-type as a possibility, or would it be better to return a fixed `application/octet-stream` if we aren't able to determine a different content type?

Reply:
Based on [Ander's comment here](https://chat.zulip.org/#narrow/stream/3-backend/topic/ignored_parameters_unsupported/near/1415199) in the related CZO conversation, I think that makes sense.

Updated the PR for that change and included a comment describing that we fallback to "application/octet-stream" when unable to guess the content-type from the filename.